aten/src/ATen/native/cpu/ReduceAllOpsKernel.cpp

```diff
@@ -108,9 +108,114 @@ static void max_all_kernel_impl(Tensor& result, const Tensor& input) {
   }
 }
 
+// For operation not support in avx/avx2
+template <typename scalar_t, typename func_t1, typename func_t2>
+inline void reduce_all_impl_two_outputs(
+    Tensor& output1,
+    Tensor& output2,
+    const Tensor& input,
+    const std::pair<scalar_t, scalar_t> ident_v,
+    func_t1 reduce_chunk_func,
+    func_t2 reduce_acc_func) {
+  using scalar_t_pair = std::pair<scalar_t, scalar_t>;
+  const int64_t input_numel = input.numel();
+  auto input_data = input.data_ptr<scalar_t>();
+  scalar_t_pair result = at::parallel_reduce(0, input_numel, internal::GRAIN_SIZE, ident_v,
+    [&](int64_t start, int64_t end, const scalar_t_pair ident) -> scalar_t_pair {
+      scalar_t_pair partial_out(ident);
+      for (int64_t i = start; i < end; i++) {
+         partial_out = reduce_chunk_func(partial_out, input_data[i]);
+      }
+      return partial_out;
+    },
+    reduce_acc_func
+  );
+  output1.fill_(result.first);
+  output2.fill_(result.second);
+}
+
+template <typename scalar_t, typename func_t, typename vec_func_t1, typename vec_func_t2>
```

Comment:
nit: vec_func_t1 and vec_func_t2 are the same type, so you don't need 2 template args here

Reply:
hmm, so I tried this a few days ago and the compiler wasn't happy.  Looks like it would be some non-trivial extra work to go from lambdas to functions which resolve to the same templated type (context: https://stackoverflow.com/questions/7477310/why-cant-i-create-a-vector-of-lambdas-of-the-same-type-in-c11).  Thoughts on if it's worth it?

Reply:
No, if it's nontrivial then it's fine to leave as is.

Reply:
sounds good.  thanks for the review!